snap-geometry-edits/src/main/java/com/esri/arcgismaps/sample/snapgeometryedits/components/MapViewModel.kt

```diff
@@ -0,0 +1,247 @@
+            if (geometryEditor.isStarted.value) {
+                createGraphic()
+            }
+        }
+        toggleButtons()
+    }
+
+    /**
+     * Creates a graphic from the geometry and add it to the GraphicsOverlay.
+     */
+    private fun createGraphic() {
+        val geometry = geometryEditor.stop()
+        val graphic = Graphic(geometry)
+
+        when (geometry!!) {
+            is Point -> graphic.symbol = GeometryEditorStyle().vertexSymbol
+            is Multipoint -> graphic.symbol = GeometryEditorStyle().vertexSymbol
+            is Polyline -> graphic.symbol = GeometryEditorStyle().lineSymbol
+            is Polygon -> graphic.symbol = GeometryEditorStyle().fillSymbol
+            is Envelope -> graphic.symbol = GeometryEditorStyle().lineSymbol
+        }
+        graphicsOverlay.graphics.add(graphic)
+        graphic.isSelected = false
+    }
+
+    /**
+     * Deletes the selected element and stops the geometry editor if there are no
+     * more elements in the geometry.
+     */
+    fun deleteSelection() {
+        if (geometryEditor.selectedElement.value != null) {
+            geometryEditor.deleteSelectedElement()
+            if (geometryEditor.geometry.value?.isEmpty == true) {
+                geometryEditor.stop()
+                toggleButtons()
+            }
+        }
+    }
+
+    /**
+     * Reverts the last event on the geometry editor.
+     */
+    fun editorUndo() {
+        geometryEditor.undo()
+    }
+
+    /**
+     * Identifies the graphic at the tapped screen coordinate in the provided [singleTapConfirmedEvent]
+     * and starts the GeometryEditor using the identified graphic's geometry. Hide the BottomSheet on
+     * [singleTapConfirmedEvent].
+     */
+    fun identify(singleTapConfirmedEvent: SingleTapConfirmedEvent) {
+        sampleCoroutineScope.launch {
+            val graphicsResult = mapViewProxy.identifyGraphicsOverlays(
+                screenCoordinate = singleTapConfirmedEvent.screenCoordinate,
+                tolerance = 10.0.dp,
+                returnPopupsOnly = false
+            ).getOrNull()
+
+            if (!geometryEditor.isStarted.value) {
+                if (graphicsResult != null) {
+                    if (graphicsResult.isNotEmpty()) {
+                        identifiedGraphic = graphicsResult[0].graphics[0]
+                        identifiedGraphic.isSelected = true
+                        identifiedGraphic.geometry?.let { geometryEditor.start(it) }
+                        toggleButtons()
+                    }
+                }
+                identifiedGraphic.geometry = null
+            }
+        }
+        dismissBottomSheet()
+    }
+
+    /**
+     * Enables or disables the UI buttons when an editing session is started or stopped.
+     */
+    private fun toggleButtons() {
+        isCreateButtonEnabled.value = !isCreateButtonEnabled.value
+        if (isCreateButtonEnabled.value) {
+            isUndoButtonEnabled.value = false
```

Comment:
Missed this one earlier - the Undo button should be enabled or disabled based on the value of `canUndo`.